File: contest-2/c++/p09-xor of all digits.cpp
// Given a string having 0's and 1's only. And it is also a palindrome too. Find the xor of all digi

#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;


int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */   
    char c;
    int x=0,y=0;
    while(cin>>c)
    {
        if(c=='1')
            y++;
    }
    x^=y%2;
    cout<<x;
    return 0;
}

File: contest-2/python/p09-xor of all digits.py
# Given a string having 0's and 1's only. And it is also a palindrome too. Find the xor of all digits.

s=input()
l=len(s)
i=0;
for j in range(l):
    if(s[j]=='1'):
        i+=1

print(i%2)
